src/barplot.ts

```diff
@@ -247,6 +247,15 @@ export function drawLabels(size:number[], data:INumericalVector, labels: d3.Sele
     'display': (rowHeight >= 15) ? 'inline' : 'none',
     'font-size': '14px'
   });
+  const categories = (<any>data.desc.value).categories;
+  const nameToLabel = new Map<string, string>();
+  if (categories) {
```

Comment:
> In this case, a cast is still needed since IValueTypeDesc neither has a categories field. Isn't it?

sure that's why I suggested the `data.desc.type === 'categorical'` check. Then you can safely convert data to `(<ICategoricalVector>data)` which then should have the proper description.